packages/react-components/src/components/resource-explorers/explorers/alarm-explorer/transformAlarmData.ts

```diff
@@ -0,0 +1,42 @@
+import { AlarmData, AlarmDataStatus } from '../../../../hooks/useAlarms';
+import { parseAlarmStateAssetProperty } from '../../../../hooks/useAlarms/transformers';
+import {
+  AlarmResource,
+  AlarmResourceWithLatestValue,
+} from '../../types/resources';
+
+export const transformAlarmData = ({
+  assetId,
+  assetModelId,
+  compositeModelId,
+  compositeModelName,
+  state,
+  status,
+}: AlarmData): {
+  resource: Partial<AlarmResourceWithLatestValue>;
+  status: AlarmDataStatus;
+} => {
+  const latestState = parseAlarmStateAssetProperty(state?.data?.at(-1));
+
+  return {
```

Comment:
This can be a follow up, but we should consider showing the state text in a different color with its icon.

Reply:
Ya, I agree, the UX hasn't been finalized, so I was leaving this as a followup so as not to redo something that might change. I can add a note on this in the code though so it's clear.